src/core/listener.c

```diff
@@ -643,6 +644,33 @@ QuicListenerClaimConnection(
         return FALSE;
     }
 
+    if (Event.NEW_CONNECTION.NewNegotiatedAlpn) {
```

Comment:
We should verify that `NewNegotiatedAlpn` actually comes from `ClientAlpnList`. Also verify that `NewNegotiatedAlpn + NewNegotiatedAlpn[0]` does not overflow `ClientAlpnList`.

Reply:
I added an address check to verify if it comes from `ClientAlpnList` via getting it's start and end address, also I added overflow check as well